### PR TITLE
Frontend form tag

### PR DESCRIPTION
**Pull request by jcderry5**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+	    // Create commentsRecord object in json form
+        
+        
+        //String json = convertToJson(commentsRecord);
+        String json = new Gson().toJson(commentsRecord);
+        
+        
+        // Send the JSON as the response
+        response.setContentType("application/json;");
+        response.getWriter().println(json);
```
Comment (KillianLeC) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:49]: This overwrites JSON pull request? Also, this feature is for frontend and not backend I thought?

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    }
+
+    /**
+    * Converts the commentsRecord instance into a JSON string using manual String concatentation.
+    */
+    private String convertToJson(PortfolioComments commentsRecord) {
```
Comment (KillianLeC) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:56]: Method not used, please remove

File: portfolio/src/main/webapp/discussion.html [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+        <form action="/data" method="POST">
+
+                <h2> Comments Below </h2>
+                <input type="text" name="user-comment">
+                <br/><br/>
+                <input type="submit" />
+        
+        </form>
+
```
Comment (KillianLeC) [inline portfolio/src/main/webapp/discussion.html:26]: nit:remove empty newlines

File: portfolio/src/main/webapp/discussion.html [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        <button onclick="generateDiscussion()">Generate Discussion!</button>
+        <div id="discussion-container"></div>
+    </div>
+  </body>
+</html>
```
Comment (KillianLeC) [inline portfolio/src/main/webapp/discussion.html:31]: end of file with newline

File: portfolio/src/main/webapp/script.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+
+// This function will handle the response and turn it to json
+function parseJSON(response){
+    console.log('Parse the response into JSON');
+
+    const discussionLog = response.json();
+
+    discussionLog.then(addToContainer);
```
Comment (KillianLeC) [inline portfolio/src/main/webapp/script.js:35]: This function is meant to parse json, but it seems that it only converts the response to json and pass that into another function. Please rename function here to indicate that it does both of these tasks. (Include into documentation for the function as well).

File: portfolio/src/main/webapp/script.js [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+// This function will add each comment to the discussion container
+function addToContainer(discussion_log){
+
+	console.log('Adding comments to the discussion-container');
```
Comment (KillianLeC) [inline portfolio/src/main/webapp/script.js:41]: nit: formatting, drop a few spaces here and align with other code

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (jcderry5) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:49]: This file looks identical to the DataServlet.java file in the previous branch. I may have made formatting changes in this file but most of my edits on this branch are front end.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+
+/** Servlet that returns some example content. TODO: modify this file to handle comments data */
+@WebServlet("/data")
+public final class DataServlet extends HttpServlet {
+
+	@Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        
+        PortfolioComments commentsRecord = new PortfolioComments();
+
```
Comment (KillianLeC) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:34]: remove new lines

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+        commentsRecord.addComment("This is momma! You got this girl :-)");
+        commentsRecord.addComment("This is dad! Go break some eggs, babygirl.");
+
+	    // Create commentsRecord object in json form
+        
+        
+        //String json = convertToJson(commentsRecord);
+        String json = new Gson().toJson(commentsRecord);
+        
+
```
Comment (KillianLeC) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:46]: remove new lines

File: portfolio/src/main/webapp/script.js [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+
+  // When the request is complete, pass the response into convertToJson()
+  discussionLogPromise.then(convertToJson);
+}
+
+/**
+* This function will convert the response to JSON and pass json
+* to the addToContainer function
+*/
+function convertToJson(response){
```
Comment (KillianLeC) [inline portfolio/src/main/webapp/script.js:30]: convertToJson

File: portfolio/src/main/webapp/script.js [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Fetches stats from the servers and adds them to the DOM.
+ */
+function generateDiscussion() {
+  // fetch() function returns a promise
+  const discussionLogPromise = fetch('/data');
+
```
Comment (KillianLeC) [inline portfolio/src/main/webapp/script.js:21]: remove new line

File: walkthroughs/week-3-server/examples/subtraction-game/src/main/webapp/script.js [hunk 11]
```diff
@@ -25,6 +25,7 @@ function getSubtractionGame() {
       totalEl.innerText = 'Total: ' + game.currentTotal;
     }
 
+	console.log(history);
```
Comment (KillianLeC) [inline walkthroughs/week-3-server/examples/subtraction-game/src/main/webapp/script.js:28]: indentation

File: walkthroughs/week-3-server/examples/server-stats/src/main/webapp/script.js [hunk 12]
```diff
@@ -19,7 +19,8 @@ function getServerStats() {
   fetch('/server-stats').then(response => response.json()).then((stats) => {
     // stats is an object, not a string, so we have to
     // reference its fields to create HTML content
-
+	console.log("hehe I'm in the console");
```
Comment (KillianLeC) [inline walkthroughs/week-3-server/examples/server-stats/src/main/webapp/script.js:22]: indentation

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 13]
```diff
@@ -26,26 +26,17 @@
...
-        String json = new Gson().toJson(commentsRecord);
-        
-        
-        // Send the JSON as the response
-        response.setContentType("application/json;");
-        response.getWriter().println(json);
-    }
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException { 
+    ArrayList<String> commentsRecord = new ArrayList<>();
```
Comment (KillianLeC) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:31]: List<String>

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 14]
```diff
@@ -26,26 +26,17 @@
...
+    // Add Sample Comments
+    commentsRecord.add("You rock! Keep on coding!");
+    commentsRecord.add("This is momma! You got this girl :-)");
+    commentsRecord.add("This is dad! Go break some eggs, babygirl.");
+	//Create commentsRecord object in json form
```
Comment (KillianLeC) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:36]: indentation plus add one new line between logical breaks.

File: portfolio/src/main/java/com/google/sps/data/PortfolioComments.java [hunk 15]
```diff
@@ -0,0 +1,43 @@
...
+package com.google.sps.data;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/** Class containing server statistics. */
+public final class PortfolioComments {
+
+	private final List<String> masterCommentList = new ArrayList<>();
+
```
Comment (tianxiao25) [inline portfolio/src/main/java/com/google/sps/data/PortfolioComments.java:24]: indentation. 

remove extra empty line. Empty line is more like a paragraph and only add when it helps readability. No need to have it between a list declaration and function used it below?

File: portfolio/src/main/java/com/google/sps/data/PortfolioComments.java [hunk 16]
```diff
@@ -0,0 +1,43 @@
...
+	public PortfolioComments() {
+        addComment("This is the first comment in masterCommentList");
```
Comment (tianxiao25) [inline portfolio/src/main/java/com/google/sps/data/PortfolioComments.java:26]: Hmm, it's a not common behavior. You can initial this list during declare time if needed. An constructor with hidden adding logic is a little bit error prone.

File: portfolio/src/main/java/com/google/sps/data/PortfolioComments.java [hunk 17]
```diff
@@ -0,0 +1,43 @@
...
+    }
+
+	public String getComment(int index){
+	    if(index >= masterCommentList.size() || index < 0){
+	        throw new IndexOutOfBoundsException("There is not a comment at the index");
+        }
+        return masterCommentList.get(index);
+    }
+
+    public int getMasterCommentListSize(){
```
Comment (tianxiao25) [inline portfolio/src/main/java/com/google/sps/data/PortfolioComments.java:40]: rename to getCommentSize?

Comment (jcderry5) [reply portfolio/src/main/java/com/google/sps/data/PortfolioComments.java:35]: Portfolio Comments has been deleted

**Pull request merged**